Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -45,44 +59,96 @@ void APSStarActor::OnLocalCharacterReady(APlayerCharacter* Character, int32 Char
+void APSStarActor::OnGameStateChanged(ECurrentGameState GameState)
+{
+	InitStarMenuAnimation();
+}
+
 // Is called when any cinematic started
 void APSStarActor::OnAnyCinematicStarted(const UObject* LevelSequence, const UObject* FromInstigator)
+{
+	SetAnimationStartTime(StartTimeHideStarsInternal);
+	TryPlayHideStarAnimation();
+}
+
+// Is called to set a start time for animations
+void APSStarActor::SetAnimationStartTime(float& StartTime)
 {
 	const UWorld* World = GetWorld();
 	check(World);
 
-	StartTimeHideStarsInternal = World->GetTimeSeconds();
+	StartTime = World->GetTimeSeconds();
 
-	// Enable tick and process this frame
+	// Enable tick for the actor
 	SetActorTickEnabled(true);
-	TryPlayHideStarAnimation();
 }
 
-// Hiding stars with animation
-void APSStarActor::TryPlayHideStarAnimation()
+// Called to initialize the Star menu animation
+void APSStarActor::InitStarMenuAnimation()
 {
-	if (!StartTimeHideStarsInternal)
+	if (ensureMsgf(UMyBlueprintFunctionLibrary::GetMyGameState()->GetCurrentGameState(), TEXT("%s: 'GetMyGameState' is null"), *FString(__FUNCTION__))
+		&& UMyBlueprintFunctionLibrary::GetMyGameState()->GetCurrentGameState() == ECurrentGameState::Menu)
 	{
-		return;
+		SetAnimationStartTime(StartTimeMenuStarsInternal);
+		TryPlayMenuStarAnimation();
+	}
+	else
+	{
+		SetActorTickEnabled(false);
+		StartTimeMenuStarsInternal = 0.f;
 	}
+}
+
+// Hiding stars with animation in main menu when cinematic is start to play
+void APSStarActor::TryPlayHideStarAnimation()
+{
+	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
+	TryPlayStarAnimation(StartTimeHideStarsInternal, CurrentRow.HideStarsAnimation);
+}
 
+// Menu stars with animation in main menu idle 
+void APSStarActor::TryPlayMenuStarAnimation()
+{
 	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
-	if (!CurrentRow.HideStarsAnimation)
+	TryPlayStarAnimation(StartTimeMenuStarsInternal, CurrentRow.MenuStarsAnimation);
+}
+
+// Playing star animation
+void APSStarActor::TryPlayStarAnimation(float& StartTime, UCurveTable* AnimationCurveTable)
+{
+	if (!StartTime || !AnimationCurveTable)
 	{
 		SetActorTickEnabled(false);
-		StartTimeHideStarsInternal = 0.f;
+		StartTime = 0.f;
 		return;
 	}
 
-	const float SecondsSinceStart = GetWorld()->GetTimeSeconds() - StartTimeHideStarsInternal;
+	const float SecondsSinceStart = GetWorld()->GetTimeSeconds() - StartTime;
 
-	const bool bIsFinished = !UGameplayUtilsLibrary::ApplyTransformFromCurveTable(this, CurrentRow.HideStarsAnimation, SecondsSinceStart);
+	const bool bIsFinished = !UGameplayUtilsLibrary::ApplyTransformFromCurveTable(this, AnimationCurveTable, SecondsSinceStart);
 	if (bIsFinished)
 	{
-		SetActorTickEnabled(false);
-		StartTimeHideStarsInternal = 0.f;
+		// if it's looped menu star animation then reset
+		// if it's one-time cinematic animation - disable tick and stop
+		if (AnimationCurveTable == UPSWorldSubsystem::Get().GetCurrentRow().MenuStarsAnimation)
```

Comment:
I like your idea to have `TryPlayStarAnimation`-helper. However, it shouldn't contain any specifics like `AnimationCurveTable == MenuStarsAnimation)
1. Make `TryPlayStarAnimation`-helper function to return bool
2. Don't compare ``AnimationCurveTable == MenuStarsAnimation)`, but always set `StartTime = 0.f`
3. in `TryPlayHideStarAnimation()`, if helper returned false, call `SetActorTickEnabled(false);`

In this way, your helper stays universal while any specific logic (like tick disable for HideStar) is handled in their corresponding functions

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/6141759e381d2b8658971efac878aaebce1a9cf7

Reply:
@h4rdmol, is not finished yet.
You have next:
```
	if (bIsFinished)
	{
		StartTimeRef = GetWorld()->GetTimeSeconds();
		return true;
	}
	else
	{
		return false;
	}
```
Where `else` will always fail.

Please see #2, where  you return `UGameplayUtilsLibrary::ApplyTransformFromCurveTable`

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/9d1eb7d37d99c460793886798dab68732855365e

Reply:
fixed a bit here: https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/8639210197ed6948fb7113ecbb6ed9372c267855